package-res/resources/web/pentaho/util/global.js

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright 2016 - 2017 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define(function() {
+
+  "use strict";
+
+  // Based on ECMAScript proposal: https://github.com/ljharb/System.global
+
+  /* globals self, window, global */
+  /* eslint no-negated-condition: 0, no-new-func: 0 */
+
+  return (typeof self !== "undefined") ? self :
+         (typeof window !== "undefined") ? window :
+         (typeof global !== "undefined") ? global :
+         Function("return this")();
```

Comment:
From [MDN, on Function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function):
"Invoking the Function constructor as a function (without using the new operator) has the same effect as invoking it as a constructor."